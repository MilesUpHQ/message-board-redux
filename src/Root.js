import React, { Component } from "react";
import { createStore, combineReducers, applyMiddleware, compose } from "redux";
import { Provider } from "react-redux";
import reduxPromise from "redux-promise";
import thunk from "redux-thunk";
import { BrowserRouter } from "react-router-dom";
import { postReducer } from "./reducers/post_reducer";

const rootReducer = combineReducers({
  posts: postReducer
});

const logger = store => next => action => {
  const prevState = store.getState();
  const result = next(action);

  console.log("[Middleware prevState]", action.type, prevState);
  console.log("[Middleware currenState]", store.getState());
  return result;
};

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const dummyState = {
  posts: [{ id: 1, title: "Magesh" }, { id: 2, title: "Rajesh" }]
};

class Root extends Component {
  render() {
    const initialState = this.props.initialState
      ? this.props.initialState
      : dummyState;

    const store = createStore(
      rootReducer,
      initialState,
      composeEnhancers(applyMiddleware(logger, reduxPromise, thunk))
    );

    return (
      <Provider store={store}>
        <BrowserRouter>{this.props.children}</BrowserRouter>
      </Provider>
    );
  }
}

export default Root;
