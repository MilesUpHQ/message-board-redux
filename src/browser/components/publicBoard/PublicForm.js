import React, { Component } from "react";

class PublicForm extends Component {
  render() {
    return (
      <div>
        <textarea className="form-control" />
        <button className="btn btn-primary">Post</button>
      </div>
    );
  }
}

export default PublicForm;
