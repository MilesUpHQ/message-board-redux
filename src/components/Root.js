import React, { Component } from "react";
import { createStore, combineReducers, applyMiddleware } from "redux";
import { Provider } from "react-redux";
import reduxPromise from "redux-promise";
import { BrowserRouter } from "react-router-dom";
import { postReducer } from "../reducers/post_reducer";

// const logger = store => next => action => {
//   const result = next(action);
//   console.log("Logger:", action.type, store.getState());
//   return result;
// };

const logger = store => {
  return next => {
    return action => {
      const result = next(action);
      console.log("Logger:", action.type, store.getState());
      return result;
    };
  };
};
const rootReducer = combineReducers({
  posts: postReducer
});

class Root extends Component {
  render() {
    const initialState = this.props.initialState ? this.props.initialState : {};
    const store = createStore(
      rootReducer,
      initialState,
      applyMiddleware(logger, reduxPromise)
    );

    return (
      <div>
        <Provider store={store}>
          <BrowserRouter>{this.props.children}</BrowserRouter>
        </Provider>
      </div>
    );
  }
}

export default Root;
