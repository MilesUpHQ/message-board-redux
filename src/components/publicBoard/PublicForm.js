import React, { Component } from "react";
import { connect } from "react-redux";

import * as actions from "../../actions/action";

class PublicForm extends Component {
  state = {
    inputText: ""
  };

  handleChange = event => {
    this.setState({
      inputText: event.target.value
    });
  };

  render() {
    return (
      <div>
        <textarea
          onChange={this.handleChange}
          value={this.state.inputText}
          className="form-control"
        />
        <button
          onClick={() => this.props.addPost(this.state.inputText)}
          className="btn btn-primary"
        >
          Post
        </button>
      </div>
    );
  }
}

export default connect(
  null,
  actions
)(PublicForm);
