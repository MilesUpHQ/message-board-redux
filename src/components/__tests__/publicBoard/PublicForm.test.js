import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import PublicForm from "../../publicBoard/PublicForm";
import Root from "../../../Root";

let wrapper;

beforeEach(() => {
  wrapper = mount(
    <Root>
      <PublicForm />
    </Root>
  );
});

afterEach(() => {
  wrapper.unmount();
});

it("should render a textarea", () => {
  expect(wrapper.find("textarea").length).toEqual(1);
});

it("should render a blue button", () => {
  expect(wrapper.find("button.btn-primary").length).toEqual(1);
});

describe("textarea on Change", () => {
  beforeEach(() => {
    wrapper.find("textarea").simulate("change", {
      target: { value: "Type something" }
    });
    wrapper.update();
  });

  it("should change value when user is typing", () => {
    expect(wrapper.find("textarea").prop("value")).toEqual("Type something");
  });

  it("should empty textarea on submit", () => {
    wrapper.find("button").simulate("click");
    wrapper.update();
    //console.log(wrapper.find("textarea").props());
    expect(wrapper.find("textarea").prop("value")).toBe("");
  });

  it("check snapshot for PublicForm", () => {
    wrapper = renderer.create(
      <Root>
        <PublicForm />
      </Root>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  // it("snapshot PublicForm component", () => {
  //   wrapper = renderer.create(
  //     <Root>
  //       <PublicForm />
  //     </Root>
  //   );
  //   expect(wrapper.toJSON()).toMatchSnapshot();
  // });
});
