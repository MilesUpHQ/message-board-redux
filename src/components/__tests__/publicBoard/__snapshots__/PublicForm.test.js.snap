// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`textarea on Change check snapshot for PublicForm 1`] = `
<div>
  <textarea
    className="form-control"
    onChange={[Function]}
    value=""
  />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Post
  </button>
</div>
`;
